### DISCUSSION TITLE
Problem with submenu

### DISCUSSION THREAD

**Discussion by steve6375** [a1ive/grub2-filemanager#341] [Q&A]
I am having a problem with using **submenu** to run a grubfm command.
Here is a test case:

If I make a /boot/grubfm/config file with

**config**
`configfile  (hd0,2)/boot/grubfm/config.txt`

which runs config.txt

**config.txt**
```
set timeout=3
set default=0
submenu "a1ive grub2 File Manager (/_ISO/MAINMENU)" --class=dir {
		grubfm "(hd0,msdos1)/_ISO/MAINMENU/"
	}
```

Then grubfm does not run **on timeout** but does run if I manually select the entry, but if I change the 'submenu' to 'menuentry' then it runs OK on timeout.

I don't seem to get this issue with other submenu items. It looks like grubfm is being selected on timeout and it is executed, but it just returns back to the menu.

Any ideas how to fix it so that it runs on timeout?
[submenu_test.zip](https://github.com/a1ive/grub2-filemanager/files/5936779/submenu_test.zip)

Thanks

**Comment by a1ive** [comment]
why use `submenu`? it's not correct.

**Reply by steve6375** [reply]
To work around the problem of setting default to a non-zero value.
I want to have the first menu and be able to set any menu item as the default.

If you set the default to say 2, then grubfm always uses a default of 2 in its file manager menu.
But if I use submenu entries, I can set the default to 0 before calling grubfm and it works.

**Reply by steve6375** [reply]
e.g. File manager runs with default set as 0
```
set timeout=3
set default=1
submenu "a1ive grub2 File Manager (/_ISO/MAINMENU)" --class=dir {
		set default=0
		grubfm "(hd0,msdos1)/_ISO/MAINMENU/"
	}
submenu "a1ive grub2 File Manager (/_ISO/MAINMENU)" --class=dir {
		set default=0
		grubfm "(hd0,msdos1)/_ISO/MAINMENU/"
	}
submenu "a1ive grub2 File Manager (/_ISO/MAINMENU)" --class=dir {
		set default=0
		grubfm "(hd0,msdos1)/_ISO/MAINMENU/"
	}
```

**Reply by a1ive** [reply]
oh, i see.
i think we may create a dummy menu config file in (proc), e.g. `(proc)/grubfm.cfg`, so users could load this menu to enter grubfm main menu.

**Comment by steve6375** [comment]
Can't grubfm always set default to 0 whenever it is invoked?

**Reply by a1ive** [reply]
It is not as easy as you say.
